testcases/kernel/syscalls/getsockopt/getsockopt01.c

```diff
@@ -1,208 +1,88 @@
-	PF_INET, SOCK_STREAM, 0, SOL_SOCKET, SO_OOBINLINE, &optval,
-		    &optlen, (struct sockaddr *)&fsin1, sizeof(fsin1),
-		    -1, EBADF, setup0, cleanup0, "bad file descriptor"}
-	, {
-	PF_INET, SOCK_STREAM, 0, SOL_SOCKET, SO_OOBINLINE, &optval,
-		    &optlen, (struct sockaddr *)&fsin1, sizeof(fsin1),
-		    -1, ENOTSOCK, setup0, cleanup0, "bad file descriptor"}
-	,
-#ifndef UCLINUX
-	{
-	PF_INET, SOCK_STREAM, 0, SOL_SOCKET, SO_OOBINLINE, 0, &optlen,
-		    (struct sockaddr *)&fsin1, sizeof(fsin1), -1,
-		    EFAULT, setup1, cleanup1, "invalid option buffer"}
-	, {
-	PF_INET, SOCK_STREAM, 0, SOL_SOCKET, SO_OOBINLINE, &optval, 0,
-		    (struct sockaddr *)&fsin1, sizeof(fsin1), -1,
-		    EFAULT, setup1, cleanup1, "invalid optlen"}
-	,
-#endif /* UCLINUX */
-	{
-	PF_INET, SOCK_STREAM, 0, 500, SO_OOBINLINE, &optval, &optlen,
-		    (struct sockaddr *)&fsin1, sizeof(fsin1), -1,
-		    EOPNOTSUPP, setup1, cleanup1, "invalid level"}
-	, {
-	PF_INET, SOCK_STREAM, 0, IPPROTO_UDP, SO_OOBINLINE, &optval,
-		    &optlen, (struct sockaddr *)&fsin1, sizeof(fsin1),
-		    -1, EOPNOTSUPP, setup1, cleanup1, "invalid option name"}
-	, {
-	PF_INET, SOCK_STREAM, 0, IPPROTO_UDP, SO_OOBINLINE, &optval,
-		    &optlen, (struct sockaddr *)&fsin1, sizeof(fsin1),
-		    -1, EOPNOTSUPP, setup1, cleanup1,
-		    "invalid option name (UDP)"}
-	, {
-	PF_INET, SOCK_STREAM, 0, IPPROTO_IP, -1, &optval, &optlen,
-		    (struct sockaddr *)&fsin1, sizeof(fsin1), -1,
-		    ENOPROTOOPT, setup1, cleanup1, "invalid option name (IP)"}
-	, {
-	PF_INET, SOCK_STREAM, 0, IPPROTO_TCP, -1, &optval, &optlen,
-		    (struct sockaddr *)&fsin1, sizeof(fsin1), -1,
-		    ENOPROTOOPT, setup1, cleanup1, "invalid option name (TCP)"}
-,};
-
-int TST_TOTAL = sizeof(tdat) / sizeof(tdat[0]);
-
-int main(int argc, char *argv[])
-{
-	int lc;
-
-	tst_parse_opts(argc, argv, NULL, NULL);
-
-	setup();
-
-	for (lc = 0; TEST_LOOPING(lc); ++lc) {
-		tst_count = 0;
-		for (testno = 0; testno < TST_TOTAL; ++testno) {
-			tdat[testno].setup();
-
-			TEST(getsockopt(s, tdat[testno].level,
-					tdat[testno].optname,
-					tdat[testno].optval,
-					tdat[testno].optlen));
-			if (TEST_RETURN != tdat[testno].retval ||
-			    (TEST_RETURN < 0 &&
-			     TEST_ERRNO != tdat[testno].experrno)) {
-				tst_resm(TFAIL, "%s ; returned"
-					 " %ld (expected %d), errno %d (expected"
-					 " %d)", tdat[testno].desc,
-					 TEST_RETURN, tdat[testno].retval,
-					 TEST_ERRNO, tdat[testno].experrno);
-			} else {
-				tst_resm(TPASS, "%s successful",
-					 tdat[testno].desc);
-			}
-			tdat[testno].cleanup();
-		}
-	}
-	cleanup();
-
-	tst_exit();
-}
+	{.sockfd = &sock_bind, .level = IPPROTO_UDP, .optname = SO_OOBINLINE, .optval = &optval,
+	.optlen = &optlen, .experrno = EOPNOTSUPP, .desc = "invalid option name"},
 
-void setup(void)
-{
-	TEST_PAUSE;
+	{.sockfd = &sock_bind, .level = IPPROTO_UDP, .optname = SO_OOBINLINE, .optval = &optval,
+	.optlen = &optlen, .experrno = EOPNOTSUPP, .desc = "invalid option name (UDP)"},
```

Comment:
Fix from the original test: this one I would describe as `"not supported option name (UDP)"`. Because errno is different from the last two and `SO_OOBINLINE` is valid name (unlike `-1`), just not supported).